backpropagation/network/neural_network.py

```diff
@@ -56,14 +61,15 @@ def _feedforward(self, x: np.array) -> np.array:
                 The output values of the neural network for the given input.
 
         """
-        for weight in self.weights:
-            x = self.activation_function.calculate_value(
-                weight @ np.insert(x, 0, 1)
-            )
+        calculate_activation = self.activation_function.calculate_value
+        x = reduce(
```

Comment:
Maybe `output` instead of `x`? Then you could use `x` instead of `z`. Just convention in lambdas for me.

Reply:
Fixed in ec76e254beec8723a922814b2492ace1fa938ed9